### PR TITLE
fix: specify distribution key for fs source

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

* **bug**: in prev impl, the distribution key is missing and we apply all state changes to vnode0 in the runtime. This behavior leads to only one fs fetch exec will work (not introducing correctness problem but low efficiency). 
  * The root cause is that the fs fetch exec will do range scan on assigned vnodes but all data is written to one vnode. 
* **fix**: set the distribution key in the frontend before writing to catalog. Do NOTHING to existing jobs, preventing data consistency problem. 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/stream_source.rs [hunk 1]
```diff
@@ -75,7 +75,7 @@ impl StreamNode for StreamSource {
             source_id: source_catalog.id,
             source_name: source_catalog.name.clone(),
             state_table: Some(
-                generic::Source::infer_internal_table_catalog()
+                generic::Source::infer_internal_table_catalog(false)
```
Comment (wenym1) [inline src/frontend/src/optimizer/plan_node/stream_source.rs:78]: May add doc on why the dist key is not required for the stream source case.

**Review by wenym1 — APPROVED** [review]

File: src/frontend/src/optimizer/plan_node/stream_source.rs (hunk 1 above)
Comment (tabVersion) [reply src/frontend/src/optimizer/plan_node/stream_source.rs:80]: done

**Review by st1page — APPROVED** [review]
LGTM, btw, what is the meaning of the partition_id in FSFetcher? Does S3 support to fetch the file names with partition?

**Comment by tabVersion** [comment]
> LGTM, btw, what is the meaning of the partition_id in FSFetcher? Does S3 support to fetch the file names with partition?

We see each file as a partition in file source. So we use file names as state table's pk and distribution key.

**Pull request merged**